backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -359,6 +363,8 @@ public EndpointDTO createEndpoint(
         @RequestBody(required = true)   EndpointDTO endpointDTO
     ) {
         if (this.backendConfig.isKesselBackendEnabled()) {
+            this.kesselAssets.createIntegration(sec, UUID.randomUUID().toString(), UUID.randomUUID().toString());
+
             this.kesselAuthorization.hasPermissionOnResource(sec, WorkspacePermission.INTEGRATIONS_CREATE, ResourceType.WORKSPACE, WORKSPACE_ID_PLACEHOLDER);
 
             return this.internalCreateEndpoint(sec, endpointDTO);
```

Comment:
Couple of things to think about here:

- These are in the wrong order – you want to make sure the user is authorized before calling inventory.
- In light of this, we are thinking about having inventory perform this permissions check on clients' behalf. In that case you would only need to call createIntegration, and you would get a 403 if not allowed. You wouldn't need to make a second call (`hasPermissionOnResource`)

Reply:
Actually, looks like you have a `createIntegration` call inside `internalCreateEndpoint`. This is a better spot for it IMO.

Reply:
Yes, the first one was a test to try the connection to the inventory. I've removed it and kept the call in the `internal` method. Once the inventory API does the authorization for us we can also remove the call to the relationships API.